rulette-core/src/main/java/com/github/kislayverma/rulette/core/ruleinput/value/defaults/InputDateValue.java

```diff
@@ -11,10 +11,10 @@
     private static final long serialVersionUID = 5666450390675442878L;
 
     private final Date value;
-    private static final DateTimeFormatter formatter = DateTimeFormat.forPattern("yyyyMMdd");
+    private static final DateTimeFormatter formatter = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
 
     public InputDateValue (String value) throws Exception {
-        this.value = value == null || value.isEmpty() ? null : formatter.parseDateTime(value).toDate();
+        this.value = (value == null || value.isEmpty()) ? null : formatter.parseDateTime(value.substring(0, 19)).toDate(); //Taking till seconds value of the date string.
```

Comment:
This should not be in the default implementation. You should implement your own InputDateValue with a specific format and use that at the time of rule system instantiation.